experimental/entity/entity.proto

```diff
@@ -0,0 +1,219 @@
+    bool hasSecureKeys = 6;
+
+	// The entity store does not extend the base EntityEnvelope -- this is typical for
+	// non-object-model formats like images (png, svg, etc)
+    bool isRaw = 7;
+
+    // For raw content types, this is set as an HTTP header
+    string contentType = 8;
+}
+
+//---------------------------------------------------------
+// Resource service enables HTTP-style requests over gRPC.
+//---------------------------------------------------------
+
+message GetEntityRequest {
+    // Path encodes the UID + kind.  The schema registry will know how to map suffix to registered kinds
+    string path = 1; 
+
+    // Version specifies which version of the 
+    string version = 2;  
+
+    // Include storage metadata in the response
+    bool withStorageMeta = 3;
+
+    // Include the entity payload in response
+    bool withPayload = 4;
+
+    // Augment results with access restrictions
+    bool withACL = 5;
+
+    // Include a list of alternative versions (support depends on storage engine) 
+    bool withPRs = 6;
+}
+
+message ListFolderRequest {
+    string path = 1; 
+
+    string filterKind = 2;
+
+    // Include storage metadata for each item
+    bool withStorageMeta = 3;
+
+    // Include the entity payload for each item
+    bool withPayload = 4;
+
+    // Optional next token to iterate with
+    string pageToken = 5; 
+    
+    // max items in the list
+    int32 maxResults = 6; 
+}
+
+message GetHistoryRequest {
+    string path = 2; 
+    string pageToken = 4;  
+    int32 maxResults = 5; 
+}
+
+message SaveEntityRequest {
+    string path = 1; 
+
+    bytes payload = 2;
+
+    string comment = 3; 
+
+    // When set, this will implement optimistic locking
+    string previousVersion = 4;  
+}
+
+message CreatePullRequest {
+    // Path to the entity we want to change
+    string path = 1; 
+
+    // Title for the new change requests
+    string title = 2;
+
+    // Comment in the new change request
+    string comment = 3; 
+
+    // Proposed payload for the enitty
+    bytes payload = 4;
+}
+
+message DeleteEntityRequest {
+    // encodes uid + kind
+    string path = 1; 
+
+    // Version may only be supported for pull requests
+    string version = 2; 
+
+    // Recursive only supported for folders
+    bool recursive = 3;
+}
+
+message DeleteResponse {
+    bool ok = 1; 
+}
+
+message ListKindsRequest {}
+message ListKindsResponse {
+    repeated KindSummary kinds = 1;
+}
+
+service EntityStore {
+    rpc GetEntity(GetEntityRequest) returns (EntityMessage);
+    rpc ListFolder(ListFolderRequest) returns (FolderListing); 
+    rpc SaveEntity(SaveEntityRequest) returns (EntityMessage); 
+    rpc DeleteEntity(DeleteEntityRequest) returns (DeleteResponse);
+    rpc GetEntityHistory(GetHistoryRequest) returns (EntityHistoryResponse);
+    rpc CreatePR(CreatePullRequest) returns (EntityMessage); 
+
+    rpc ListKinds(ListKindsRequest) returns (ListKindsResponse); 
```

Comment:
Do we also need a `RegisterKind` here?

Reply:
Not in the protobuf contract -- that will need to be supported by the underlying system